clients/src/main/java/org/apache/kafka/common/utils/Time.java

```diff
@@ -30,7 +30,40 @@
  */
 public interface Time {
 
-    Time SYSTEM = new SystemTime();
+    //  A time implementation that uses the system clock and sleep call.
+    //  Use inline implementation to ensure that only one Time#SYSTEM exists in a program
+    Time SYSTEM = new Time() {
+        @Override
+        public long milliseconds() {
+            return System.currentTimeMillis();
+        }
+
+        @Override
+        public long nanoseconds() {
+            return System.nanoTime();
+        }
+
+        @Override
+        public void sleep(long ms) {
+            Utils.sleep(ms);
+        }
+
+        @Override
+        public void waitObject(Object obj, Supplier<Boolean> condition, long deadlineMs) throws InterruptedException {
```

Comment:
It seems this method is used by `ProducerMetadata#awaitUpdate`. Maybe we can have a separate PR to do a bit refactor for it?

Reply:
Thank you very much for your reply, but I don't fully understand which method you want to refactor.
The developer of `Time#waitObject()` seems to want to provide a unified synchronization method based on `Time`, but it doesn't seem to need to be refactored.
But `ProducerMetadata#awaitUpdate` can remove the `synchronized` keyword

Reply:
I open https://issues.apache.org/jira/browse/KAFKA-16937 for it. We can have more discussion there